OsmAnd/src/net/osmand/plus/measurementtool/MeasurementEditingContext.java

```diff
@@ -0,0 +1,282 @@
+package net.osmand.plus.measurementtool;
+
+import android.util.Pair;
+import android.view.View;
+import android.widget.ProgressBar;
+
+import net.osmand.Location;
+import net.osmand.data.LatLon;
+import net.osmand.plus.ApplicationMode;
+import net.osmand.plus.GPXUtilities.TrkSegment;
+import net.osmand.plus.GPXUtilities.WptPt;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.OsmandSettings;
+import net.osmand.plus.activities.MapActivity;
+import net.osmand.plus.routing.RouteCalculationParams;
+import net.osmand.plus.routing.RoutingHelper;
+import net.osmand.plus.views.Renderable;
+import net.osmand.router.RouteCalculationProgress;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+
+public class MeasurementEditingContext {
+
+	private MapActivity mapActivity;
+
+	private TrkSegment before = new TrkSegment();
+	// cache should be deleted if after changed or snappedToRoadPoints
+	private TrkSegment beforeCacheForSnap;
+
+	private TrkSegment after = new TrkSegment();
+	// cache should be deleted if after changed or snappedToRoadPoints
+	private TrkSegment afterCacheForSnap;
+
+	private boolean inMovePointMode;
+	private boolean inAddPointBeforeMode;
```

Comment:
This variables pobably needs to be deleted "inAddPointBeforeMode", "inAddPointAfterMode" - doesn't seem to be usable